sidecar/sidecar.go

```diff
@@ -172,11 +150,7 @@ func Run(conf config.Config) error {
 		}
 	}
 
-	if conf.Supervise {
-		appSupervisor.DoAppSupervision(conf.App, registrationAgent)
-	} else {
-		select {}
-	}
+	appSupervisor.DoAppSupervision(registrationAgent)
```

Comment:
The reason is that `DoAppSupervision()` does more than just supervising child processes: it also intercepts signals (e.g. SIGTERM due to `docker stop`), and stops the registration agent (so that it can deregister).